### PR TITLE
Prepare 0.65.1 release

### PR DESCRIPTION
**Pull request by tigrannajaryan**
- Update versions.yaml to 0.65.1
- Retract v0.65.0
- Add 0.65.1 to CHANGELOG.md and note that v0.65.0 is broken

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
The failing Windows build should be fixed by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16423

**Comment by tigrannajaryan** [comment]
Rebased to include fix https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16423

**Comment by tigrannajaryan** [comment]
Added `retract v0.65.0` to go.mod

**Comment by tigrannajaryan** [comment]
Added 0.65.1 to CHANGELOG.md and noted that v0.65.0 is broken

**Comment by tigrannajaryan** [comment]
deb and rpm builds are failing. Is this expected?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,13 @@
 
 <!-- next version -->
 
+## v0.65.1
+
+### 🧰 Bug fixes 🧰
+
+- Fixes a problem with v0.65.1 which contained incorrect component versions. v0.65.0 is
```
[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply CHANGELOG.md]: Fixed, thank you.

**Review by TylerHelmuth — COMMENTED** [review]
@tigrannajaryan the 0.65.0 changelog also need updated with some of these changelog entries: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/.chloggen.

**Review by codeboten — APPROVED** [review]

**Comment by codeboten** [comment]
@tigrannajaryan looks like the rpm/deb failure was caused by the setup-ruby action being deprecated. i've committed a change to update it to use `ruby/setup-ruby` instead

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan the 0.65.0 changelog also need updated with some of these changelog entries: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/.chloggen.

Done.

**Comment by Aneurysm9** [comment]
Each and every `go.mod` file here will require its own `retract v0.65.0` statement.

**Comment by Aneurysm9** [comment]
> > Each and every `go.mod` file here will require its own `retract v0.65.0` statement.
> 
> Can you clarify why is it necessary? The (sub) modules are perfectly usable, we just don't want the Collector to be published.

[`exporter/datadogexporter/go.mod`](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16422/files#diff-c79f7c731b2b8a5e11f31c5d7e1da85550586ad25d22d0dba105a6ac5df0a44c), to take just one example, has numerous references to `v0.64.0` modules and must be assumed to be invalid.

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan looks like the rpm/deb failure was caused by the setup-ruby action being deprecated. i've committed a change to update it to use `ruby/setup-ruby` instead

@codeboten that didn't seem to help.

**Comment by tigrannajaryan** [comment]
> > > Each and every `go.mod` file here will require its own `retract v0.65.0` statement.
> > 
> > 
> > Can you clarify why is it necessary? The (sub) modules are perfectly usable, we just don't want the Collector to be published.
> 
> [`exporter/datadogexporter/go.mod`](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16422/files#diff-c79f7c731b2b8a5e11f31c5d7e1da85550586ad25d22d0dba105a6ac5df0a44c), to take just one example, has numerous references to `v0.64.0` modules and must be assumed to be invalid.

@Aneurysm9 I added retract to all modules.

**Comment by tigrannajaryan** [comment]
@codeboten please ping me when you have a fix for deb/rpm so that I can rebase this.

**Comment by dmitryax** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16436 was failing due to the 0.65.0 release not the GH actions. This PR's build should pass once rebased

**Comment by tigrannajaryan** [comment]
Rebased.

**Review by bogdandrutu — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
@dmitryax no luck

**Comment by TylerHelmuth** [comment]
@tigrannajaryan @dmitryax @codeboten I _think_ the issue is that the `set_release_tag.sh` script is no longer working correctly after https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16389. 

The result of that script isn't in the output
image

but the next step in the workflow shows that the version isn't getting properly set for the next script.

image

The workflow calls the script with `./internal/buildscripts/packaging/fpm/${{ matrix.package_type }}/build.sh "${{ steps.github_tag.outputs.tag }}" "amd64" "./dist/"`

If `steps.github_tag.outputs.tag` isn't set, then the script does:

```bash
if [[ -z "$VERSION" ]]; then
    latest_tag="$( git describe --abbrev=0 --match v[0-9]* )"
    VERSION="${latest_tag}-post"
fi
```

Those line match the screenshot above, so I think that means that $VERSION isn't getting set, which implies `steps.github_tag.outputs.tag` is not set.

**Comment by TylerHelmuth** [comment]
Nevermind, the last release does the same thing: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3428200998/jobs/5712744869. :man-shrugging:

**Comment by TylerHelmuth** [comment]
It looks like the script always ends up using `git describe --abbrev=0 --match v[0-9]*` to get the latest tag so maybe the issue is that since 0.65.0 is the latest tag and since 0.65.0 is broken the build doesn't work ?

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-contrib-maintainer I haven't work on the contrib for a while, I don't know how the release works and don't have the expertise to solve this myself. I will need your help to figure it out.

**Comment by dmitryax** [comment]
I'm looking into this

@TylerHelmuth I see that at least [this commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/8cea4e1f3910724a1136f8026b2f934ad7bc8ce8) after 0.65.0 partially succeeded, so not sure it's related to 0.65.0.

**Comment by bogdandrutu** [comment]
This commit correctly should be done against the release branch

**Comment by dmitryax** [comment]
@bogdandrutu I can make another PR against `release/0.65.x`. The problem is that we don't have such branch, only `release/0.65.0`. Do you think we should proceed on main or create an additional `release/0.65.x` branch from 0.65.0 release commit?

**Comment by bogdandrutu** [comment]
we have now, https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/release/v0.65.x, I renamed that branch

**Comment by tigrannajaryan** [comment]
Closed in favour of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16457

**Pull request closed without merging**